src/useFetch.ts

```diff
@@ -70,31 +71,25 @@ function useFetch<TData = any>(...args: UseFetchArgs): UseFetch<TData> {
         path,
         method,
         theController,
-        cachePolicy,
         cacheLife,
         cache,
         routeOrBody,
         body,
         interceptors.request
       )
-
+      
       if (!suspense) setLoading(true)
       error.current = undefined
 
-      if (response.isCached && cachePolicy === CACHE_FIRST) {
-        if (response.isExpired) {
-          cache.delete(response.id)
-          cache.delete(response.ageID)
-        } else {
-          try {
-            res.current.data = await tryGetData(response.cached, defaults.data)
-            data.current = res.current.data as TData
-            if (!suspense) setLoading(false)
-            return data.current
-          } catch (err) {
-            error.current = err
-            setLoading(false)
-          }
+      if (response.isCached && (persist || cachePolicy === CACHE_FIRST)) {
```

Comment:
I don't think we need the `persist` here. This would mean NO_CACHE or another cachePolicy without a cache would pass here if `persist` is true.